### DISCUSSION TITLE
Question on .http file format and compatibility

### DISCUSSION THREAD

**Discussion by KevTheRev13** [AnWeber/httpyac#522] [General]
Hello. 

First, thank you for your work on httpyac. Awesome tool.

I'd like to learn more about the .http file format and the syntax it uses. Where/who exactly is defining the .http file syntax ? Is there a centralized definition or governing body ?

I ask because I'm wondering just how interoperable httpyac-written .http files are with other programs that use .http files.
Examples being Microsoft Visual Studio (https://learn.microsoft.com/en-us/aspnet/core/test/http-files?view=aspnetcore-8.0), IntelliJ (https://www.jetbrains.com/help/idea/http-client-in-product-code-editor.html), and standalone projects (example https://marketplace.visualstudio.com/items?itemName=humao.rest-client) .

Can an .http file written in one be run in another ? (assuming they implement all specs of the standard)

Thank you

**Comment by AnWeber** [comment]
Yes, but actually no: [https://github.com/JetBrains/http-request-in-editor-spec/blob/master/spec.md](https://github.com/JetBrains/http-request-in-editor-spec/blob/master/spec.md)

In my opinion, [vscode.rest-client](https://github.com/Huachao/vscode-restclient/graphs/code-frequency) (2016) was first released. The specification of [Jetbrains](https://www.jetbrains.com/help/idea/http-client-in-product-code-editor.html) was created in 2017 and already has deviations from the implementation in vscode.rest-client. I created httpyac in 2020 with the goal to build a superset of both, allow Extensibility even without the author and implement some more features I was missing in vscode.rest-client ([NodeJS PR :-)](https://github.com/Huachao/vscode-restclient/pull/674)). Visual Studio support of http files is new to me. There are some other clients using http Files like [dothttp](https://docs.dothttp.dev/) and many Forks (vscode.rest-client is not activly developed since three years).

Visual Studio seems to be a subset of vscode.rest-client. Intellij and vsocde.rest-client has some overlapping features. httpyac is a superset of both, but unfortunately I screwed up in a few places. I believe httpyac is compatible with Intellij, but to vscode.rest-client I have discrepancies. 

~I will add a detailed table with my knowledge, but it will take some time.~ I'm not sure it's going to do any good. There are many differences and often in details. The basis for simple requests is similar, and from then on the deviations start. For example accessing the `response` is possible in vscode.rest-client directly, in Jetbrains only via response handler script and in my case both variants, but the response object unfortunately differs from vscode.rest-client. I use NodeJS to access properties and vscode.rest-client parses the property accessor. Websocket/ Grpc and EventStreaming is only possible in Intellij and httpyac. Using `@ref` and `@import` is only supported in httpyac, but the idea is a draft from vscode.rest-client, but the draft changed since then. The only thing I can guarantee you is that Intellij and httpyac are identical if you use the subset of Intellij. I pay attention to that.

**Comment by KevTheRev13** [comment]
Oh that is very interesting. Thank you for the information.

With Postman making more changes to their software that one could view as less data privacy friendly (requiring cloud usage more aggressively) - I believe there is an increased personal and enterprise interest in local tools like httpyac. I see opportunity for .http file usage as a whole.

I would love to see a long term goal of httpyac trying to be "mostly" compatible with Microsoft's implementation. Right now, it's very basic - but they are adding features and I suspect it will take off. More and more people will get exposure to .http files in Visual Studio, and httpyac has the opportunity to be a better / more flexible / non MS solution here that I think others would be interested in.

This is all long term speculation though. In my own testing, they currently are broadly compatible - but I haven't tried anything too advanced.

**Reply by AnWeber** [reply]
@KevTheRev13 I will try to update httpyac according to Visual Studio, but it is not my tech bubble. If you find some discrepancies just create an Issue to let me know of it.

**Reply by KevTheRev13** [reply]
Great, thank you. Will do.
